src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +248,40 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+int verify_crl_signature(X509_CRL* crl, X509* issuer) {
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return 1;
+  }
+  int ret = X509_CRL_verify(crl, ikey);
+  EVP_PKEY_free(ikey);
+  return ret;
+}
+
+int verify_crl_cert_issuer_names_match(X509_CRL* crl, X509* issuer) {
+  return X509_NAME_cmp(X509_get_issuer_name(issuer), X509_CRL_get_issuer(crl));
```

Comment:
Please check that both `X509_get_issuer_name(issuer)` and `X509_CRL_get_issuer(crl)` are not null before passing them into `X509_NAME_cmp`, otherwise this will cause a segfault.

Reply:
Done